image-customize

```diff
@@ -52,91 +51,101 @@ def prepare_install_image(base_image, install_image, resize):
     return install_image
 
 
-def run_command(machine_instance, commandlist):
-    """Run command inside image"""
-    for foo in commandlist:
+class ActionBase(argparse.Action):
+    '''Keep an ordered list of actions'''
+
+    def __call__(self, parser, namespace, value, option_string=None):
+        if 'actions' not in namespace:
+            setattr(namespace, 'actions', [])
```

Comment:
I feel like this could be avoided by using the `default=[]` and `dest='actions'` parameters to `.add_argument()`.  Then `self.dest` here will be set to `action` and you can just say `getattr(namespace, self.dest).append(...)`

It would also be much neater that this class doesn't hardcode its destination name, and would let you remove the check for making sure something exists at the destination when you go to run the actions.

Reply:
Right, it avoids the [gs]etattr, but hardcoding the dest name was kind of the point.. I could drop it as you say, but then at least dest= has to be repeated. default= doesn't have to be, it just looks a little asymmetric. But at this point I don't care much either way, changed accordingly.